src/DynamoPackages/Package.cs

```diff
@@ -528,15 +528,45 @@ internal void MarkForUninstall(IPreferences prefs)
             {
                 prefs.PackageDirectoriesToUninstall.Add(RootDirectory);
             }
-            RaisePropertyChanged(nameof(LoadState.ScheduledState));
+            RaisePropertyChanged(nameof(LoadState));
         }
 
         internal void UnmarkForUninstall(IPreferences prefs)
         {
             LoadState.ResetScheduledState();
 
             prefs.PackageDirectoriesToUninstall.RemoveAll(x => x.Equals(RootDirectory));
-            RaisePropertyChanged(nameof(LoadState.ScheduledState));
+            RaisePropertyChanged(nameof(LoadState));
+        }
+
+        internal void MarkForLoad(IPreferences prefs)
+        {
+            if (BuiltInPackage)
+            {
+                LoadState.SetScheduledForLoad();
+
+                if (prefs.PackageDirectoriesToUninstall.Contains(RootDirectory))
+                {
+                    prefs.PackageDirectoriesToUninstall.Remove(RootDirectory);
+                }
+                RaisePropertyChanged(nameof(LoadState));
+            }
+        }
+
+        internal void UnmarkForLoad(IPreferences prefs)
+        {
+            if (BuiltInPackage && 
+                LoadState.ScheduledState == PackageLoadState.ScheduledTypes.ScheduledForLoad && 
+                LoadState.State == PackageLoadState.StateTypes.Unloaded)
+            {
+                LoadState.ResetScheduledState();
+                if (!prefs.PackageDirectoriesToUninstall.Contains(RootDirectory))
+                {
+                    prefs.PackageDirectoriesToUninstall.Add(RootDirectory);
```

Comment:
I remember we had some trouble with the new package path views in preferences dialog being updated when this model was changed - I think thats fixed now, right @sm6srw ?

Reply:
Yes